web3.md

```diff
@@ -637,5 +651,61 @@ WEB3 JSON RPC
+      requires isString( #getJSON("key", J) )
+       andBool notBool #addrFromPrivateKey(#getString("key", J)) in ACCTS
+
+    rule <k> #firefly_addAccount => #sendResponse( "result": false ) ... </k>
+         <params> [ ({ _ } #as J), .JSONList ] </params>
+          <activeAccounts> ACCTS </activeAccounts>
+      requires isString( #getJSON("key", J) )
+       andBool #addrFromPrivateKey(#getString("key", J)) in ACCTS
+
+    rule <k> #firefly_addAccount =>  #sendResponse( "error": {"code": -32025, "message":"Method 'firefly_addAccount' has invalid arguments"} ) ... </k> [owise]
+
+    syntax KItem ::= "#loadAccountData" JSON
+ // ----------------------------------------
+    rule <k> #loadAccountData J => . ... </k>
+         <account>
+           <acctID>  ACCTID  </acctID>
+           <balance>     ( _ => #if            isString( #getJSON("balance",  J) ) #then #parseHexWord(#getString("balance", J)) #else 100                    #fi )     </balance>
+           <code>        ( _ => #if            isString( #getJSON("code",     J) ) #then #parseByteStack(#getString("code",  J)) #else .ByteArray:AccountCode #fi )        </code>
+           <nonce>       ( _ => #if            isString( #getJSON("nonce",    J) ) #then #parseHexWord(#getString("nonce",   J)) #else 0                      #fi )       </nonce>
+           <storage>     ( _ => #if notBool isJSONUndef( #getJSON("storage",  J) ) #then #parseMap(#getJSON("storage",       J)) #else .Map                   #fi )     </storage>
+           <origStorage> ( _ => #if notBool isJSONUndef( #getJSON("storage",  J) ) #then #parseMap(#getJSON("storage",       J)) #else .Map                   #fi ) </origStorage>
+         </account>
+      requires isString( #getJSON("address", J) )
+       andBool #parseHexWord(#getString("address", J)) ==Int ACCTID
+
+    rule <k> #loadAccountData J => . ... </k>
+         <account>
+           <acctID>  ACCTID  </acctID>
+           <balance>     ( _ => #if            isString( #getJSON("balance",  J) ) #then #parseHexWord(#getString("balance", J)) #else 100                    #fi )     </balance>
+           <code>        ( _ => #if            isString( #getJSON("code",     J) ) #then #parseByteStack(#getString("code",  J)) #else .ByteArray:AccountCode #fi )        </code>
+           <nonce>       ( _ => #if            isString( #getJSON("nonce",    J) ) #then #parseHexWord(#getString("nonce",   J)) #else 0                      #fi )       </nonce>
+           <storage>     ( _ => #if notBool isJSONUndef( #getJSON("storage",  J) ) #then #parseMap(#getJSON("storage",       J)) #else .Map                   #fi )     </storage>
+           <origStorage> ( _ => #if notBool isJSONUndef( #getJSON("storage",  J) ) #then #parseMap(#getJSON("storage",       J)) #else .Map                   #fi ) </origStorage>
+         </account>
+      requires isString( #getJSON("key", J) )
+       andBool #addrFromPrivateKey(#getString("key", J)) ==Int ACCTID
```

Comment:
This will turn into quite an inefficient map lookup, since we're computing the key in the side-condition. So instead of looking up exactly the account to load data into, we need to cycle through all the accounts and check "is this the right account".

Better to directly look up on the `ACCTID`, instead of constraining it in a side-condition.